### ISSUE TITLE
Error ssl check

### ISSUE THREAD

**Issue by fansubmaniac** [blind-coder/rcmcarddav#292]
Hello,
I try to connect cardav plugin to my nextcloud, install worked fine, i can fill all informations in the rouncube > cardav parameters but when validate, nothing is saved.
In the log file i found some error 

`Jun  6 11:40:32 serveur roundcube: <6rg2ujfk> DISCOVERY: (retrieve_addressbooks) SEARCHING https://XXXXXXXXXXXXXX:443/remote.php/dav/addressbooks/users/xxxxx/contacts/ (Depth: 0)
Jun  6 11:40:32 serveur roundcube: <6rg2ujfk> DISCOVERY: (cdfopen) retrieve_addressbooks requesting https://XXXXXXXXXXXXXX:443/remote.php/dav/addressbooks/users/xxxxx/contacts/ as user michael [RL 5]
Jun  6 11:40:32 serveur roundcube: error:1408F10B:SSL routines:ssl3_get_record:wrong version number
Jun  6 11:40:32 serveur roundcube: PHP Fatal error:  Uncaught Httpful\Exception\ConnectionErrorException: Unable to connect to "https://XXXXXXXXXXXXXX:443/remote.php/dav/addressbooks/users/xxxxx/contacts/": 35 error:1408F10B:SSL routines:ssl3_get_record:wrong version number in /var/www/roundcube/vendor/nategood/httpful/src/Httpful/Request.php:1060#012Stack trace:#012#0 /var/www/roundcube/vendor/nategood/httpful/src/Httpful/Request.php(229): Httpful\Request->buildResponse(false)#012#1 /var/www/roundcube/plugins/carddav/carddav_common.php(185): Httpful\Request->send()#012#2 /var/www/roundcube/plugins/carddav/carddav_discovery.php(126): carddav_common->cdfopen('https://cloud.k...', Array, Array)#012#3 /var/www/roundcube/plugins/carddav/carddav_discovery.php(96): carddav_discovery->retrieve_addressbooks('/remote.php/dav...', Array)#012#4 /var/www/roundcube/plugins/carddav/carddav.php(544): carddav_discovery->find_addressbooks('https://cloud.k...', 'michael', '{BASE64}SmNANDg...')#012#5 /var/www/roundcube/program/lib/Roundcube/rcube_plugin_api.php(469): carddav->cd_save(Array)#012#6 /va in /var/www/roundcube/vendor/nategood/httpful/src/Httpful/Request.php on line 1060
`
It appear that ssl check failed but i can't find any information to disable it un roundcube (except for smtp and imap) I use letsencrypt on both with latest version of each app on ubuntu 18.04. 

Thanks for help.

**Comment by mstilkerich** [comment]
Hello,
would you mind trying the v4 branch?

See installation instructions here: https://github.com/blind-coder/rcmcarddav/tree/v4

**Comment by fansubmaniac** [comment]
Hello,
Thanks for your answer. I've updated to V4 branch wich gave me the same error (in the frontend this time):
`CURLE_SSL_CONNECT_ERROR 35 wrong version number`
I was thinking maybe my curl version was outdating but not. So I googlize (again) and I've found [this topic](https://github.com/guzzle/guzzle/issues/2593). The last post gave me the solution : my vhost in apache  was set with IP and not with the hostname. Now it's working fine :-)
Thank's again.
